code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -17,58 +17,566 @@
+
+	if (!damage)
+		return
+
+	damagetaken += damage
+
+	if(!isbroken)
+		if (user)
 			user.do_attack_animation(src)
-			playsound(get_turf(src), 'sound/effects/Glassbr3.ogg', 75, 1)
-			isbroken = 1
-			density = 0
-			icon_state = "pylon-broken"
-			set_light(0)
+		if(prob(damagetaken*0.5))
+			shatter()
 		else
-			user << "You hit the pylon!"
+			if (user && !ranged)
+				user << "You hit the pylon!"
 			playsound(get_turf(src), 'sound/effects/Glasshit.ogg', 75, 1)
 	else
-		if(prob(damage * 2))
-			user << "You pulverize what was left of the pylon!"
+		if(prob(damagetaken))
+			if (user)
+				user << "You pulverize what was left of the pylon!"
 			qdel(src)
-		else
+		else if (user && !ranged)
 			user << "You hit the pylon!"
 		playsound(get_turf(src), 'sound/effects/Glasshit.ogg', 75, 1)
 
+	start_process()
+
+/obj/structure/cult/pylon/proc/shatter()
+	visible_message(
+	"<span class='danger'>The pylon shatters into shards of crystal!</span>",
+	"You hear a tinkle of crystal shards"
+	)
+	playsound(get_turf(src), 'sound/effects/Glassbr3.ogg', 75, 1)
+	isbroken = 1
+	if (pylonmode == 2)
+		//If the pylon had a soul in it then it plays a creepy evil sound as the soul is released
+		var/possibles = list('sound/hallucinations/far_noise.ogg',
+		'sound/hallucinations/growl3.ogg',
+		'sound/hallucinations/veryfar_noise.ogg',
+		'sound/hallucinations/wail.ogg',
+		'sound/voice/hiss5.ogg')
+		playsound(get_turf(src), pick(possibles), 50, 1)
+
+	pylonmode = 0 //A broken pylon loses its soul. Even if repaired it will need a new sacrifice to re-empower it
+
+	//Make sure we stop it from firing
+	target = null
+	sacrificer = null
+	notarget = 0
+	last_target_loc = null
+	process_interval = 1
+	//It will stop processing after the next check, start of the next process
+
+	empowered = 0
+	density = 0
+	update_icon()
+
+
 
 /obj/structure/cult/pylon/proc/repair(mob/user as mob)
 	if(isbroken)
-		user << "You repair the pylon."
+		if(user)
+			user << span("notice","You weave forgotten magic, summoning the shards of the crystal and knitting them anew, until it hovers flawless once more.")
 		isbroken = 0
 		density = 1
-		icon_state = "pylon"
-		set_light(5)
+	else if (damagetaken > 0)
+		user << span("notice","You meld the crystal lattice back into integrity, sealing over the cracks until they never were.")
+
+	else
+		user << span("notice","The crystal lights up at your touch")
+
+	process_interval = 1 //Wake up the crystal
+	notarget = 0
+	damagetaken = 0
+	update_icon()
+
+
+/obj/structure/cult/pylon/update_icon()
+	overlays.Cut()
+	if (pylonmode == 2)
+		anchored = 1
+		if (empowered > 0)
+			overlays += "crystal_overcharge"
+			set_light(7, 3, l_color = "#a160bf")
+		else
+			set_light(6, 3, l_color = "#3e0000")
+			overlays += "crystal_turret"
+	else if (!isbroken)
+		set_light(5, 2, l_color = "#3e0000")
+		overlays += "crystal_idle"
+		if (pylonmode == 1)
+			anchored = 1
+		else
+			anchored = 0
+	else
+		anchored = 0
+		set_light(0)
+
+
+

```

Comment:
Mind removing some of this whitespace? Doesn't really contribute much to readability.